### PR TITLE
[QC-443] Represent Checks configuration as C++ structures

### PR DESCRIPTION
**Pull request by knopers8**
Includes also PostProcessingTaskSpec and ExternalTaskSpec, but these are only
used by CheckRunners so far.

Not the last PR for this issue...

### CODE REVIEW

File: Framework/include/QualityControl/CheckConfig.h [hunk 1]
```diff
@@ -30,9 +33,12 @@ struct CheckConfig {
   std::string className;
   std::string detectorName = "MISC"; // intended to be the 3 letters code;
   std::unordered_map<std::string, std::string> customParameters = {};
-  std::string policyType = "OnAny";
-  std::vector<std::string> objectNames;
+  UpdatePolicyType policyType = UpdatePolicyType::OnAny;
+  std::vector<std::string> objectNames{}; // fixme: if object names are empty, allObjects are true, consider reducing to one var
```
Comment (Barthelemy) [inline Framework/include/QualityControl/CheckConfig.h:37]: good point

File: Framework/include/QualityControl/runnerUtils.h [hunk 2]
```diff
@@ -121,11 +122,37 @@ inline int computeRunNumber(const framework::ServiceRegistry& services, int defa
...
+
+inline std::string computePassName(const std::string& fallbackPassName = "")
+{
+  std::string passName;
+  passName = fallbackPassName;
+  ILOG(Debug, Devel) << "Period Name returned by computePassName : " << passName << ENDM;
+  return passName;
+}
+
+inline std::string computeProvenance(const std::string& fallbackProvenance = "")
```
Comment (Barthelemy) [inline Framework/include/QualityControl/runnerUtils.h:148]: I think that there is a rebase that is missing because these lines were added in another PR.

**Review by Barthelemy — APPROVED** [review]

File: Framework/include/QualityControl/runnerUtils.h (hunk 2 above)
Comment (knopers8) [reply Framework/include/QualityControl/runnerUtils.h:148]: i will check

Comment (knopers8) [reply Framework/include/QualityControl/runnerUtils.h:148]: I do not understand what is the problem. For now there are two copies of `compute*` functions until we stop using the ones which require the whole ptree (aggregators still use them).

Comment (Barthelemy) [reply Framework/include/QualityControl/runnerUtils.h:148]: oh, sorry, I had misunderstood what was going on. All good !

**Pull request merged**